fix: rune popup for v11.18. use lockfile to locate port & token.

magicwenli opened the pull request:
test ok

cangzhang commented on src/share/file.ts:
```diff
@@ -81,39 +80,23 @@ export const getLatestLogFile = async (dir: string) => {
-    const port = url.match(/:(\d+)$/)?.[1] ?? null;
+    const lockfile = await fs.readFile(`${lockfilePath}`, 'utf8');
+    const port = lockfile.split(`:`)[2];
+    const passwd = lockfile.split(`:`)[3];
```
I think it's better to rename the variable `passwd` to `token`, due to it's more a token than a password

magicwenli replied:
cool

Sokwva replied:
good!

cangzhang commented on src/share/file.ts:
```diff
@@ -81,39 +80,23 @@ export const getLatestLogFile = async (dir: string) => {
 export const getLcuToken = async (dirPath: string) => {
   const appendGameToDir = config.get(`appendGameToDir`);
-  const dir = `${appendGameToDir ? `${dirPath}/Game` : dirPath}/Logs/LeagueClient Logs`;
+  const lockfilePath = `${appendGameToDir ? `${dirPath}/LeagueClient` : dirPath}/lockfile`;
```
Coz lcu of different servers have different folder structure, I will verify it later for NA lcu.

cangzhang reviewed (commented):
Good work! 👍

magicwenli commented:
cool

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Al Cheung ***@***.***>
Sent: Thursday, September 9, 2021 12:46:09 PM
To: cangzhang/champ-r ***@***.***>
Cc: magicWenli ***@***.***>; Author ***@***.***>
Subject: Re: [cangzhang/champ-r] fix: rune popup for v11.18. use lockfile to locate port & token. (#89)


@cangzhang commented on this pull request.

Good work! 👍

________________________________

In src/share/file.ts<https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcangzhang%2Fchamp-r%2Fpull%2F89%23discussion_r704947229&data=04%7C01%7C%7Cbc0fbe8b70bf4b27514708d9734cbece%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637667595726932317%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=kHPaF%2Br68hC6AzKTh4ylpC69FXNU3zMZ52KHdnvILw8%3D&reserved=0>:

> -        break

-      }

-      continue

-    }

-

-    if (!content) {

-      console.error(`[LCU] cannot find target renderer log.`)

-      return [null, null, null]

-    }

-

-    const url = content.match(/https(.*)\/index\.html/)?.[1] ?? ``;

-    const token = url.match(/riot:(.*)@/)?.[1] ?? null;

-    const port = url.match(/:(\d+)$/)?.[1] ?? null;

+    const lockfile = await fs.readFile(`${lockfilePath}`, 'utf8');

+    const port = lockfile.split(`:`)[2];

+    const passwd = lockfile.split(`:`)[3];



I think it's better to rename the variable passwd to token, due to it's more a token than a password

________________________________

In src/share/file.ts<https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcangzhang%2Fchamp-r%2Fpull%2F89%23discussion_r704947695&data=04%7C01%7C%7Cbc0fbe8b70bf4b27514708d9734cbece%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637667595726932317%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=%2FGrqhmvC67Dq5TVviMK7g4Rf9%2BQVVwTxzzr7pK6sWv8%3D&reserved=0>:

>

 export const getLcuToken = async (dirPath: string) => {

   const appendGameToDir = config.get(`appendGameToDir`);

-  const dir = `${appendGameToDir ? `${dirPath}/Game` : dirPath}/Logs/LeagueClient Logs`;

+  const lockfilePath = `${appendGameToDir ? `${dirPath}/LeagueClient` : dirPath}/lockfile`;



Coz lcu of different servers have different folder structure, I will verify it later for NA lcu.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcangzhang%2Fchamp-r%2Fpull%2F89%23pullrequestreview-749820847&data=04%7C01%7C%7Cbc0fbe8b70bf4b27514708d9734cbece%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637667595726942276%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=yUhtmL%2FicGpxPY0dFVU9kGbxfF6y3FY6JMspRyAa%2FQU%3D&reserved=0>, or unsubscribe<https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALPROIW7WW63LNWIWJOANGLUBA3ZDANCNFSM5DWGFIBQ&data=04%7C01%7C%7Cbc0fbe8b70bf4b27514708d9734cbece%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637667595726942276%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=p7VZksvbArTfZ558ZXuC6cHC4nCbIhhYy5FT6XMTEGc%3D&reserved=0>.

cangzhang commented on src/share/file.ts:
```diff
@@ -81,39 +80,23 @@ export const getLatestLogFile = async (dir: string) => {
+    const port = lockfile.split(`:`)[2];
+    const token = lockfile.split(`:`)[3];
+
+    const url = `://riot:${passwd}@127.0.0.1:${port}`
```
Looks like this one was forgot. https://github.com/cangzhang/champ-r/runs/3552301364

magicwenli replied:
thanks for reviewing 🥵

magicwenli commented:
I have an idea for a new feature, adding an automatic acceptance switch. But I am a newbie to Github, I wrote the function in the change/auto-accept branch under magicwenli/champ-r. Now all changes to my main branch will be reflected in this pull request, which is unexpected. 

I don’t know how to merge the latest changes in my repo without affecting this pull request.

btw, do you like this idea?

cangzhang commented:
well you have to close this PR and make commits in your own repo.

cangzhang commented:
I will cherry pick your commits and make a new PR coz I have to release a new version later tonight. Try ideas in your own repo and PR is always welcome🤓

magicwenli commented:
appreciate that! 🥰
